### ISSUE TITLE
[AnyPrecision optimizer] Kahan compensation buffer should be stored in state dict for checkpointing

### ISSUE THREAD

**Issue by lessw2020** [pytorch/torchdistx#58]
Problem - if the user runs AnyPrecision optimizer with Kahan and checkpoints the model/optimizer, restarting training will start with an empty compensation buffer. 
This is not a blocking problem, but would lower the user's accuracy during training for the next 1 -2 epochs while the compensation buffer was refilled with updated values. 

Solution - add state dict entry for compensation buffer, and unit test to verify start/stop training scenario.
